src/Http/Middleware/CsrfProtectionMiddleware.php

```diff
@@ -241,13 +241,17 @@ protected function _validateToken(ServerRequestInterface $request): void
         $body = $request->getParsedBody();
         if (is_array($body) || $body instanceof ArrayAccess) {
             $post = Hash::get($body, $this->_config['field']);
-            if ($this->_compareToken($post, $cookie)) {
+            if (Security::constantEquals($post, $cookie)) {
+                return;
+            } else if ($this->_compareToken($post, $cookie)) {
                 return;
             }
```

Comment:
```suggestion
            // Allow non-hardened tokens until 4.1.0
            if (Security::constantEquals($post, $cookie)) {
                return;
            }
            // Validates hardened tokens generated since 4.0.6
            // The above check will circumvent this until 4.1.0
            if ($this->_compareToken($post, $cookie)) {
                return;
            }
```